### PR TITLE
3256 - Add factor and floor validations

### PR DESCRIPTION
**Pull request by sonaisrayel**

### CODE REVIEW

File: api/rest/floor.go [hunk 1]
```diff
@@ -59,61 +58,56 @@ func FloorGetAllHandler(c *fiber.Ctx) error {
...
+		return utils.ErrorResponse(c, fiber.StatusBadRequest, fmt.Sprintf("Failed to update metadata table for floor, %s", err))
 	}
 
-	return c.Status(http.StatusOK).JSON(Response{Status: "ok", Message: "Floor and metadata tables successfully updated"})
+	return utils.SuccessResponse(c, fiber.StatusOK, "Floor and Metadata tables successfully updated")
 }
 
-func updateFloorMetaData(c *fiber.Ctx, data *FloorUpdateRequest) string {
-	utils.ConvertingAllValues(data)
+func updateFloorMetaData(c *fiber.Ctx, data *FloorUpdateRequest) error {
```
Comment (finkiel) [inline api/rest/floor.go:82]: this function shouldn't be in API module but in core

File: api/rest/factor.go [hunk 2]
```diff
@@ -63,110 +58,54 @@ func FactorGetAllHandler(c *fiber.Ctx) error {
...
-		log.Error().Err(err).Str("body", string(c.Body())).Msg("Failed to insert metadata update to queue")
-		return "failed to insert metadata update to queue"
+		return utils.ErrorResponse(c, fiber.StatusInternalServerError, "Failed to insert metadata update to queue")
 	}
-	return ""
+
+	return nil
 }
 
 func updateFactor(c *fiber.Ctx, data *FactorUpdateRequest) error {
```
Comment (finkiel) [inline api/rest/factor.go:111]: this function should be in core module not in api

File: api/rest/factor.go [hunk 3]
```diff
@@ -63,110 +58,54 @@ func FactorGetAllHandler(c *fiber.Ctx) error {
...
-		c.Status(http.StatusBadRequest)
-		return nil, true
-	}
-
-	if data.Device == "" {
-		c.SendString(fmt.Sprintf("Device is mandatory"))
-		c.Status(http.StatusBadRequest)
-		return nil, true
-	}
+func updateMetaData(c *fiber.Ctx, data *FactorUpdateRequest) error {
```
Comment (finkiel) [inline api/rest/factor.go:81]: the function should be in core module

File: api/rest/floor.go [hunk 4]
```diff
@@ -59,61 +58,56 @@ func FloorGetAllHandler(c *fiber.Ctx) error {
...
-		log.Error().Err(err).Str("body", string(c.Body())).Msg("Failed to insert metadata update to queue for floor")
-		return "failed to insert metadata update to queue for floor"
+		return utils.ErrorResponse(c, fiber.StatusInternalServerError, "Failed to insert metadata update to queue")
 	}
-	return ""
+
+	return nil
 }
 
 func updateFloors(c *fiber.Ctx, data *FloorUpdateRequest) error {
```
Comment (finkiel) [inline api/rest/floor.go:113]: please move the function to core module

File: api/rest/factor.go (hunk 2 above)
Comment (sonaisrayel) [reply api/rest/factor.go:111]: done

**Pull request merged**